lib/compute/operation.js

```diff
@@ -0,0 +1,143 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*!
+ * @module compute/operation
+ */
+
+'use strict';
+
+/**
+ * @type {module:common/util}
+ * @private
+ */
+var util = require('../common/util.js');
+
+/**
+ * Create an Operation object to interact with a Google Compute Engine
+ *     operation.
+ *
+ * @constructor
+ *
+ * @throws {Error} if an operation name is not provided.
+ *
+ * @param {module:compute} scope - The scope of the operation, can be either a
+ *     `Compute` object, a `Zone` object or a `Region` object.
+ * @param {string} name - Operation name.
+ * @param {object=} metadata - Operation metadata.
+ *
+ * @example
+ * var gcloud = require('gcloud')({
+ *   keyFilename: '/path/to/keyfile.json'
+ * });
+ *
+ * var compute = gcloud.compute();
+ * var myZone = compute.zone('example-zone');
+ * var myRegion = compute.region('example-region');
+ *
+ * var globalOperation = compute.operation('global-operation');
+ * var regionOperation = myRegion.operation('region-operation');
+ * var zoneOperation = myZone.operation('zone-operation');
+ */
+function Operation(scope, name, metadata) {
+  this.scope = scope;
+  this.name = name;
+  this.metadata = metadata;
+
+  if (!util.is(this.name, 'string')) {
+    throw new Error('A name is needed to use a Compute Engine Operation.');
+  }
+}
+
+/**
+ * Delete the operation.
+ *
+ * @param {function} callback - The callback function.
+ *
+ * @example
+ * operation.delete(function(err) {});
+ */
+Operation.prototype.delete = function(callback) {
+  callback = callback || util.noop;
+  this.makeReq_('DELETE', '', null, true, callback);
+};
+
+/**
+ * Get the operation's metadata. For a detailed description of metadata see
+ * [Operation resource](https://goo.gl/sWm1rt).
+ *
+ * @param {function=} callback - The callback function.
+ *
+ * @example
+ * operation.getMetadata(function(err, metadata, apiResponse) {});
+ *
+ * //-
+ * // To check the status of an operation and its progress `getMetadata` in
+ * // combination with `setTimeout` can be used.
+ * //-
+ * var checkStatus = function() {
+ *   operation.getMetadata(function(err, metadata, apiResponse) {
+ *     if (err) {
+ *       // An error occurred
+ *     } else {
+ *       if (metadata.status === 'DONE') {
+ *         // the operation completed
+ *         // `metadata.error` is set with errors if the operation failed
+ *         // `metadata.warnings` possibly contains warnings
+ *       } else {
+ *         // `metadata.progress` is a progress indicator in [0,100]
+ *         setTimeout(checkStatus, 10);
+ *       }
+ *     }
+ *   });
+ * };
+ *
+ * setTimeout(checkStatus, 10);
+ */
+Operation.prototype.getMetadata = function(callback) {
+  callback = callback || util.noop;
+  var self = this;
+  this.makeReq_('GET', '', null, null, function(err, resp) {
+    if (err && (!resp || resp.name !== self.name)) {
```

Comment:
The JSON that comes back when you do get an operation entity contains an `error` field that causes the core of `gcloud-node` to think that the request failed with an error. With that check I am checking that we actually have an error and not an operation entity: we have an error if the response is empty or the name field in the response is not the one of the operation we are trying to get.  
This is workaround but I think it makes more sense than changing how the core of `gcloud-node` requests works which is already very good IMHO.